### PR TITLE
Fix a configuration key error in forward() of MusicgenForConditionalGeneration

### PR DESCRIPTION
**Pull request by IntelliNik**
Hi everyone,

I think I noticed a bug in the forward function of MusicgenForConditionalGeneration. When calculating the loss with given labels, I get the error "'MusicgenConfig' object has no attribute 'vocab_size'" as only the decoder.config has a vocab_size entry. 

I think this should be the correct way to implement the loss calculation.

### CODE REVIEW

**Comment by amyeroberts** [comment]
Gentle ping @sanchit-gandhi

**Comment by amyeroberts** [comment]
Gentle ping @sanchit-gandhi @ylacombe

**Comment by ylacombe** [comment]
Hey @IntelliNik, thanks for opening this issue! 

Currently, Musicgen doesn't support training, and that the current loss computation is wrong, but I've opened #29802 to address this. Feel free to get inspired from this if you need to do training right now but otherwise, I hope it'll be merged in the coming days!

**Comment by ylacombe** [comment]
#29802 has been merged, you should be able to correctly compute the loss now! You can also refer to this [repo](https://github.com/ylacombe/musicgen-dreamboothing) for an example of how to fine-tune it.

**Comment by ylacombe** [comment]
https://github.com/huggingface/transformers/pull/29802 has been merged, you should be able to correctly compute the loss now! You can also refer to this [repo](https://github.com/ylacombe/musicgen-dreamboothing) for an example of how to fine-tune it.
